Fixing dict conversion bug

benjefferies opened the pull request:
### What is the context of this PR?
Describe what you have changed and why, link to other PRs or Issues as appropriate.

When posting invalid answers that contain questions with skip_conditions
state modification causes a error to be raised which redirects the user
to a server error page when they should see the errors.
We now create a deep copy of the object so when we change state it
doesn't affect the state item.

### How to review 
Describe the steps required to test the changes (include screenshots if appropriate).
See steps to replicated in #538

ajmaddaford commented on app/parser/v0_0_1/schema_parser.py:
```diff
@@ -251,7 +248,7 @@ def _parse_question(self, schema, questionnaire):
             question.title = ParserUtils.get_required_string(schema, "title")
             question.description = ParserUtils.get_required_string(schema, "description")
-            question.skip_condition = self._parse_skip_condition(ParserUtils.get_optional(schema, "skip_condition"))
+            question.skip_condition = schema.get("skip_condition")
```
I don't think we should be changing how one property of the schema object works, this would make it inconsistent with every other property. Is it possible to revert this and add an as_dict method to `schema/skip_condition.py`?

benjefferies replied:
I'd agree it's against the grain of how the schema is now but knowing that we are going to be removing the schema model imminently to use a dict anyway is it worth it also it's only being used in that one place?

benjefferies committed:
Fixing dict conversion bug

When posting invalid answers that contain questions with skip_conditions
state modification causes a error to be raised which redirects the user
to a server error page when they should see the errors.
We call copy_as_dict to convert to a copied dict

ajmaddaford commented on app/schema/skip_condition.py:
```diff
@@ -1,5 +1,11 @@
     def __init__(self):
         self.when = None
+
+    def copy_as_dict(self):
```
Do we need to know it's a copy, or can we just call the method `as_dict`?

benjefferies replied:
I would question why they weren't using .**dict** if I saw it that's all

jonnyshaw89 replied:
Why does it need to be a copy? Are we modifying it? if so, should we be?
